packages/next/build/utils.ts

```diff
@@ -1133,3 +1134,13 @@ export function isFlightPage(
   })
   return isRscPage
 }
+
+export function getUnresolvedModuleFromError(
+  error: string
+): string | undefined {
+  const moduleErrorRegex = new RegExp(
+    `Module not found: Can't resolve '(\\w+)'`
+  )
+  const [, moduleName] = error.match(moduleErrorRegex) || []
+  return builtinModules.find((item: string) => item === moduleName)
```

Comment:
```suggestion
  return builtinModules.includes(moduleName)
```

But still, this isn't reliable enough. I think we should make this on hold and have more discussions.

Reply:
`find` will return the module name or undefined, which will be used later. includes only returns boolean

Reply:
cc @javivelasco do you think if we can make this more general to be reusable for all middleware functions?